### ISSUE TITLE
Disabling eslint in styled components

### ISSUE THREAD

**Issue by kuworking** [eslint/eslint#12153]
I'm trying to disable `eslint` so that doesn't complain in `vscode` with `prettier` way of structuring code, but I can't (testing all possible solutions with `eslint-disable` and also with `prettier-ignore`, no matter what I do I cannot get rid of the warning)

This is a `GatsbyJS` component that uses `Styled Components`

```
    "lerna": "^3.16.4",
    "babel-eslint": "^10.0",
    "eslint": "^6.2.1",
    "eslint-config-prettier": "^6.1.0",
    "eslint-config-standard": "^14.0.0",
    "eslint-plugin-import": "^2.16.0",
    "eslint-plugin-node": "^9.1.0",
    "eslint-plugin-prettier": "^3.0.1",
    "eslint-plugin-promise": "^4.1.1",
    "eslint-plugin-react": "^7.12.4",
    "eslint-plugin-standard": "^4.0.0",
    "gulp": "^4.0.0",
    "gulp-help": "^1.6.1",
    "prettier": "^1.16.4"
```

![image](https://user-images.githubusercontent.com/4474004/63586256-40423c80-c5a1-11e9-9143-a7b4f2a2adda.png)

![image](https://user-images.githubusercontent.com/4474004/63586309-54863980-c5a1-11e9-9a31-301227bb5178.png)

(with this one prettier eliminates the `eslint-enable`)

![image](https://user-images.githubusercontent.com/4474004/63586358-6d8eea80-c5a1-11e9-822c-8b61134d1768.png)

![image](https://user-images.githubusercontent.com/4474004/63586450-9c0cc580-c5a1-11e9-9102-17e4ae9e013c.png)

![image](https://user-images.githubusercontent.com/4474004/63586496-b34bb300-c5a1-11e9-87b0-64018c7383c8.png)

(with this one prettier also eliminates the `eslint` rule)

![image](https://user-images.githubusercontent.com/4474004/63586548-d1b1ae80-c5a1-11e9-8459-d7ca784c5cf4.png)

**Comment by kaicataldo** [comment]
It's difficult for us to help you without some more information:

1. What versions of ESLint, Node, and npm are you using?
1. What is the actual error you're experiencing? 
1. Does it happen when you run `eslint` from the command line or only in VS Code?
1. It looks like you're using `babel-eslint` - does this happen with the default parser as well?

Not knowing the above, it looks to me like it would be best to disable ESLint for the entire styled-component string, not just the contents of this particular expression.

**Comment by kuworking** [comment]
> It's difficult for us to help you without some more information:
> 
> 1. What versions of ESLint, Node, and npm are you using?
> 2. What is the actual error you're experiencing?
> 3. Does it happen when you run `eslint` from the command line or only in VS Code?
> 4. It looks like you're using `babel-eslint` - does this happen with the default parser as well?
> 
> Not knowing the above, it looks to me like it would be best to disable ESLint for the entire styled-component string, not just the contents of this particular expression.

1. versions
`ESLint 6.2.2`
`npm 6.90`
`Node 10.16.0`
2. The error is that I get the warnings and I'm trying to solve them (by disabling `eslint` within the minimum code possible)
3. No, I run `eslint` while in `VS Code` and doing `prettier` manually (Alt Shift K), and I see the `eslint` and `prettier` warnings in `VS Code`
4. I'd say I have no problem when compiling, my problem is with the `VS Code` warnings that I would like to disable

You mean the entire component? Many of my components have similar expressions, I'd prefer not disabling eslint there but if it's the only solution that's fair enough

**Comment by kaicataldo** [comment]
Can you please share the output of the warnings? If you could also share your configuration, that would be great. I can't tell without the information above, but if it's an issue with integrating ESLint and Prettier, [Prettier's docs](https://prettier.io/docs/en/integrating-with-linters.html) have a section about this.

**Comment by kuworking** [comment]
The warnings shows indeed a conflict with prettier, `eslint` wants an intendation that prettier doesn't respect, so my first approach was to just disable eslint for these very specific cases where both plugins seem not to cope with each other

```
Expected indentation of 4 spaces but found 6.
Expected indentation of 6 spaces but found 8.
Expected indentation of 6 spaces but found 8.
```

I've also tried to disable prettier, but I've been unlucky with both approaches

```
.eslintrc.json

{
    "env": {
      "browser": true,
      "es6": true,
      "node": true
    },
    "extends": ["prettier"],
    "parser": "babel-eslint",
    "parserOptions": {
      "ecmaFeatures": {
        "jsx": true
      },
      "ecmaVersion": 2018,
      "sourceType": "module"
    },
    "plugins": ["react", "prettier"],
    "rules": {
      "strict": 0,
      "prettier/prettier": ["error"],
      "indent": ["error", 2],
      "semi": ["error", "never"]
    }
  }

```

```
.prettierrc

{
  "semi": false,
  "singleQuote": true,
  "trailingComma": "es5",
  "printWidth": 120
}
```

Versions (vs code extensions)
```
Babel  Javascript 0.0.23
ESLint 1.9.1
MDX 0.1.0
MDX Inline 0.1.0
Prettier 1.10.0
vscode-styled-components 0.0.26
```

**Comment by kaicataldo** [comment]
You’ll need to remove the following from your configured rules, as they’re overriding the configuration set by extending off of the Prettier plugin, resulting in the reported conflict:

```js
"rules": {
    "indent": ["error", 2],
    "semi": ["error", "never"]
}
```

Prettier handles both white space and semi styling.

**Comment by kuworking** [comment]
I confirm that this solve the issue, thanks a lot!
